xmppserver/src/main/java/org/jivesoftware/openfire/nio/NettyOutboundConnectionHandler.java

```diff
@@ -113,7 +113,11 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
                 // RFC 7590 Section 3.4 writes: "In particular for XMPP server-to-server interactions, it can be reasonable
                 // for XMPP server implementations to accept encrypted but unauthenticated connections when Server Dialback
                 // keys [XEP-0220] are used." In short: if Dialback is allowed, unauthenticated TLS is better than no TLS.
-                if (SASLAuthentication.verifyCertificates(connection.getPeerCertificates(), domainPair.getRemote(), true)) {
+
```

Comment:
In line with earlier comments in this PR: I'm worried that with this code, the code will accept _any_ certificate when the 'accept-self-signed' option is enabled. We should only trust properly signed ones (as we already do without this option), and self-signed ones.

Reply:
If the remote server certificate is self signed we can add it to the local-server (acting as client) trust store for encryption, but this is not sufficient to prompt the local-server (acting as client) to send its certificate over without trying to manipulate the TLS handshake. 

True acceptance of self-signed certs can be achieved if the peers self-signed cert is added to the  trust store prior to initiating a connection (results in true encrypted + SASL S2S). 

We can arguably say that if we are encrypted and are accepting self-signed certs, then the certificate has been validated as part of the TLS handshake and so can skip authentication (effectively it looks like SASL occurs but without verifying certificates again), but this would be less secure than the current implementation.